docs/docs/custom-graph-components.mdx

```diff
@@ -0,0 +1,429 @@
+---
+id: custom-graph-components
+sidebar_label: Custom Graph Components
+title: Custom Graph Components
+abstract: You can extend Rasa Open Source with custom NLU components and policies. This page provides a guide on how to develop your own custom graph components.
+---
+
+Rasa Open Source provides a variety of [NLU components](components) and
+[policies](policies) out of the box. You can customize them or create your
+own components from scratch by using custom graph components.
+
+To use your custom graph component with Rasa Open Source it has to fulfill the following
+requirements:
+- It has to implement the [`GraphComponent` interface](#the-graphcomponent-interface)
+- It has to be [registered](#registering-components-with-the-model-configuration)
+  with the used model configuration
+- It has to be used in the [configuration file](#using-custom-components-in-your-model-configuration)
+- It has to use type annotations. Rasa Open Source makes use of the type annotations
+  to validate your model configuration.
+  [Forward references](https://www.python.org/dev/peps/pep-0484/#forward-references)
+  are not allowed. If you're using Python 3.7 you can use
+  [`from __future__ import annotations`](https://www.python.org/dev/peps/pep-0563/#enabling-the-future-behavior-in-python-3-7)
+  to get rid of forward references.
+
+
+## Graph Components
```

Comment:
@koaning Do you have a simplified graph picture for this section by any chance?

Reply:
@koaning ?

Reply:
This one? 

![image](https://user-images.githubusercontent.com/1019791/141172059-083b7c0e-af3c-4a10-aecd-bac5c79e7f83.png)

Reply:
perfect thank you!